### ISSUE TITLE
2.2.8 - Magento now ships the full quantity of an item that has been partially refunded

### ISSUE THREAD

**Issue by jameshalsall** [magento/magento2#22006]
It looks like https://github.com/magento/magento2/commit/10fb01f3702c16411808ae6a24db57ad2413ea83 has introduced a bug allowing Magento to ship the full quantity of items that have been partially refunded.

### Preconditions (*)
Magento 2.2.8, issue was not present in 2.2.7

My environment is not relevant here.

### Steps to reproduce (*)
1. Place an order with qty 2 of an item
2. Refund 1 of that item
3. Create a shipment for qty 2 of that item
4. Shipment created for qty 2 instead of qty 1

### Expected result (*)
Shipment should not be created, which was the behaviour in 2.2.7.

### Actual result (*)
Shipment is created successfully.

**Comment by magento-engcom-team** [comment]
Hi @jameshalsall. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@jameshalsall do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by jameshalsall** [comment]
@magento-engcom-team give me 2.2.8 instance

**Comment by magento-engcom-team** [comment]
Hi @jameshalsall. Thank you for your request. I'm working on Magento 2.2.8 instance for you

**Comment by magento-engcom-team** [comment]
Hi @jameshalsall, here is your [Magento instance](https://i-22006-2-2-8.instances.magento-community.engineering).
Admin access: [https://i-22006-2-2-8.instances.magento-community.engineering/admin](https://i-22006-2-2-8.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by jameshalsall** [comment]
Credit memo for 2 of the item: https://i-22006-2-2-8.instances.magento-community.engineering/admin/sales/order_creditmemo/view/creditmemo_id/1/

Shipment for all of the original item qty: https://i-22006-2-2-8.instances.magento-community.engineering/admin/admin/order_shipment/view/shipment_id/2/

**Comment by shikhamis11** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, here is your [Magento instance](https://i-22006-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-22006-2-2-develop.instances.magento-community.engineering/admin](https://i-22006-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98939` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by shikhamis11** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, here is your [Magento instance](https://i-22006-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22006-2-3-develop.instances.magento-community.engineering/admin](https://i-22006-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by shikhamis11** [comment]
this issue is not present at 2.3-develop instance

**Comment by salelsol** [comment]
After upgrading Magento from 2.2.7 to 2.2.8 when we try add a Tracking  Number to some order pushing Shipping  button then receive this issue: Shipment with id "" does not exist. Also receive email with all tracking numbers ended before.
![all_tracking_mail](https://user-images.githubusercontent.com/16081377/55200605-31588900-51bf-11e9-9d4f-558a2c62dee3.jpg)

**Comment by jameshalsall** [comment]
@shikhamis11 yeah it was not ported into 2.3 branch for some reason, it only affects 2.2 branch

**Comment by jameshalsall** [comment]
any update on this? it seems like a pretty fundamental change that is not intended because it's not present in 2.3 branch

**Comment by engcom-Alfa** [comment]
Hello @jameshalsall !

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
